draft-steele-cose-merkle-tree-proofs.md

```diff
@@ -60,72 +60,50 @@ informative:
+The extensibility of the approach is demonstrated by providing CBOR encodings for RFC9162.
 
 --- middle
 
 # Introduction
 
-Merkle trees are verifiable data structures that support secure data storage,
+Merkle trees are one of many verifiable data structures that enable tamper evident secure information storage,
 through their ability to protect the integrity of batches of documents or collections of statements.
 
-A merkle proof is a path from a leaf to a root in a merkle tree.
+Merkle trees can be constructed from simple operations such as concatenation and digest via a cryptographic hash function,
+however, more advanced constructions enable proofs of different properties of the underlying verifiable data structure.
 
-Merkle proofs can be used to prove a document is in a database (proof of inclusion),
-or that a smaller set of statements are contained in a large set of statements (selective disclosure proofs).
+Verifiable data structure proofs can be used to prove a document is in a database (proof of inclusion),
+that a database is append only (proof of consistency), that a smaller set of statements are contained
+in a large set of statements (proof of disclosure, a special case of proof of inclusion),
+or proof that certain data is not yet present in a database (proofs of non inclusion).
 
-Typically, merkle trees are constructed from simple operations such as concatenation and digest via a cryptographic hash function.
+Differences in the representation of verifiable data structures, and verifiable data structure proof types,
+can increase the burden for implementers, and create interoperability challenges for transparency services.
 
-The simple design and valuable cryptographic properties of merkle trees have been leveraged in many network and database applications.
-
-Differences in the representation of a merkle tree, merkle leaf and merkle inclusion proof can increase the
-burden for implementers, and create interoperability challenges.
-
-This document describes the three data structures necessary to use merkle proofs with COSE envelopes.
+This document describes how to convey verifiable data structures, and associated proof types in COSE envelopes.
 
 ## Requirements Notation
 
 {::boilerplate bcp14-tagged}
 
 # Terminology
 
-Leaf:
-
-: A merkle tree leaf is the cryptographic hash of a sequence of bytes that combines Leaf Payload and Extra Data.
-
-Merkle Tree:
-
-: A Merkle tree is a tree where every leaf is a cryptographic hash of a sequence of
-bytes and every node that is not a leaf is the cryptographic hash of the its child nodes.
-
-Merkle Root:
-
-: A Merkle root is the root node of a tree; this cryptographic hash is a committment to the content of the tree.
-
-Merkle Tree Algorithm:
-
-: A Merkle tree algorithm specifies how to construct the tree and how to compute its root.
-
-Leaf Payload and Extra Data:
+Verifiable Data Structure:
```

Comment:
Changes after #19 start here.

Reply:
I just approved the 'update introduction' PR separately. Might we abandon that and just squash this one once approved?

Reply:
merging subsequently seems fine to me.